src/node_port.py

```diff
@@ -151,7 +159,65 @@ def delete_unit_service(self) -> None:
             if e.status.code == 403:
                 self.on_deployed_without_trust()
                 return
-            else:
-                raise
+            raise
+
+    def _node_name(self, unit_name: str) -> str:
+        """Return the node name for this unit's pod ip."""
+        try:
+            pod = self.client.get(
+                Pod,
+                name=unit_name.replace("/", "-"),
+                namespace=self.namespace,
+            )
+        except ApiError as e:
+            if e.status.code == 403:
+                self.on_deployed_without_trust()
+                return
+
+            raise
+
+        return pod.spec.nodeName
+
+    def get_node_ip(self, unit_name: str) -> Optional[str]:
+        """Return node IP for the provided unit."""
+        try:
+            node = self.client.get(
+                Node,
+                name=self._node_name(unit_name),
+                namespace=self.namespace,
+            )
+        except ApiError as e:
+            if e.status.code == 403:
+                logger.error("Could not delete service, application needs `juju trust`")
+                self.on_deployed_without_trust()
+                return
+
+            raise
+        # [
+        #    NodeAddress(address='192.168.0.228', type='InternalIP'),
+        #    NodeAddress(address='example.com', type='Hostname')
+        # ]
+        # Remember that OpenStack, for example, will return an internal hostname, which is not
+        # accessible from the outside. Give preference to ExternalIP, then InternalIP first
+        # Separated, as we want to give preference to ExternalIP, InternalIP and then Hostname
+        for typ in ["ExternalIP", "InternalIP", "Hostname"]:
+            for a in node.status.addresses:
+                if a.type == typ:
+                    return a.address
+
+    def get_node_port(self, port_to_match: int, unit_name: str) -> int:
+        """Return node port for the provided port to match."""
+        service = self.get_unit_service(unit_name=unit_name)
+
+        if not service or not service.spec.type == "NodePort":
+            raise FailedToFindServiceError(f"No service found for port on {unit_name}")
+
+        for svc_port in service.spec.ports:
+            if svc_port.port == 27018:
```

Comment:
**suggestion:** This feels like a magic number, and coming at this blind (I'm sure it's the default Mongo port), I don't know what it's for. Would be nice as a global somewhere.